### DISCUSSION TITLE
Object tracking tools notes

### DISCUSSION THREAD

**Discussion by torydebra** [ADVRHumanoids/tpo_vision#2] [General]
- IIT human sensing and perception tool https://github.com/hsp-iit/roft
  Ok, but I should ask how to use from camera, ie live data, and not files...

- Mediapipe 3dobject detection https://google.github.io/mediapipe/solutions/objectron
  It seems only shoes, chairs, cameras, and cups are trackable

- OpenMMLab [MMEngine](https://github.com/open-mmlab/mmengine) https://github.com/open-mmlab/mmdetection3d
  As many others: is it possible to use it with live data? Pytorch based

- Various awesome lists, like https://github.com/ZhongqunZHANG/awesome-6d-object which says:
  - [CenterSnap](https://github.com/zubair-irshad/CenterSnap)
  - [NOCS](https://hughw19.github.io/NOCS_CVPR2019)
  - [BundleTrack](https://github.com/wenbowen123/BundleTrack) 
  - [se(3)-TrackNet](https://github.com/wenbowen123/iros20-6d-pose-tracking)

- Intel tutorials with ROS / opencv / tensorflow integrations. check expecially:
  - https://github.com/intel/ros_object_analytics *discontinued*
  - https://dev.intelrealsense.com/docs/tensorflow-with-intel-realsense-cameras
  - ros_object_analytics abandoned, ros2_object_analytics seems abandoned too since ros2 dashing

- ROS people developed [Object Recognition Kitchen](https://wg-perception.github.io/object_recognition_core/),
  but is it for tracking, or only detection/recognition?  
  BTW, it seems abandoned, https://github.com/wg-perception/object_recognition_core from official repos it stops at kinetic version

- https://github.com/j96w/DenseFusion

- https://manipulation.csail.mit.edu/pose.html tutorial 

- PAZ https://oarriaga.github.io/paz/. Seems very nice but no proper doc for the pose estimation...

- https://github.com/opendr-eu/opendr ~To check~
  Update: maybe interesting in future, not usable for this work now, check my issue https://github.com/opendr-eu/opendr/issues/399

**Comment by torydebra** [comment]
[Detectron2](https://github.com/facebookresearch/detectron2)

**Comment by torydebra** [comment]
https://github.com/DLR-RM/3DObjectTracking

**Comment by torydebra** [comment]
https://airvlab.github.io/grasp-anything/

**Comment by torydebra** [comment]
benchmark about manipulation learning: https://robot-colosseum.github.io/

**Comment by torydebra** [comment]
## Damiano 
- foundation pose by nvidia - l'ultimo uscito
- boundlesdf by nvidia

# From a survey paper
(survey: https://www.sciencedirect.com/science/article/pii/S1047320324001287#d1e3389)
- GraspNet
- https://github.com/luyh20/FGC-GraspNet
-

**Comment by torydebra** [comment]
From this paper : https://www.sciencedirect.com/science/article/pii/S1047320324001287#d1e3389

- https://github.com/graspnet/anygrasp_sdk Seems serious code
- FGC-GraspNet - https://github.com/luyh20/FGC-GraspNet - just one repo, but instructions about downloading it

**Comment by torydebra** [comment]
https://github.com/sunhan1997/IndusGrasp

**Comment by torydebra** [comment]
https://github.com/NVlabs/FoundationStereo
TLDR: Our method takes as input a pair of stereo images and outputs a dense disparity map, which can be converted to a metric-scale depth map or 3D point cloud.

**Comment by torydebra** [comment]
https://sites.google.com/view/visuomotor-correspondence
Self-Supervised Correspondence  in Visuomotor Policy Learning
Training to "see" object from rgbd image, showing it to the camera. The network learns features. 
Then, user select a point on the object, the robot is able to look for this point on the object, placed in different orientations, and grasp the object from such selected point.
But RGBD camera used always, on the robot wrist

**Comment by torydebra** [comment]
dexnet by berkely, how could I have missed this?
https://berkeleyautomation.github.io/dex-net/

**Comment by torydebra** [comment]
DinoV3 recently realeased, more high level feature extraction for things like segmentation, pythorch implementation
https://github.com/facebookresearch/dinov3

but there is sparse matching: https://github.com/facebookresearch/dinov3/blob/main/notebooks/dense_sparse_matching.ipynb
